### ISSUE TITLE
Expose Clipboard#read/write with data transfer

### ISSUE THREAD

**Issue by liang2zhu1** [microsoft/vscode#77790]
Can you provide HTML clipboard access which is available in [Electron](https://electronjs.org/docs/api/clipboard#clipboardreadhtmltype)? i.e. expose that in [extHostClipboard.ts](https://github.com/microsoft/vscode/blob/56693de122b1345d9df1c2c37fb2cb7a23705bf7/src/vs/workbench/api/common/extHostClipboard.ts)

**Comment by jrieken** [comment]
We have no plans of exposing any of the "custom" electron API but our plan is to explore the JS standard APIs: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard. Once read/write is ready, and supported in Electron, we will expose them in the API.

**Comment by liang2zhu1** [comment]
Thanks for the answer, sad but makes sense.

**Comment by jrieken** [comment]
I keeping this open so that we can tackle this once the underlying API is ready

**Comment by spottedmahn** [comment]
If this was implemented, would that allow for more than text in the [Clipboard API](https://code.visualstudio.com/api/references/vscode-api#Clipboard)?

Looks like this [Paste Image extension](https://github.com/mushanshitiancai/vscode-paste-image/issues/56) doesn't work in WSL b/c of this?

Remote -> Tips & Tricks -> [Clipboard does not work](https://code.visualstudio.com/docs/remote/troubleshooting#_clipboard-does-not-work)

>Unfortunately, this may cause the extension to incorrectly integrate with the clipboard on the remote side.
>Resolution: The extension can switch to the VS Code clipboard API to resolve the problem.

**Comment by mmdjiji** [comment]
But I saw https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read, mozilla standard supports read() method, which can be used to get image from the clipboard.

**Comment by mmdjiji** [comment]
> @mmdjiji
> 
> Everyone here is [already aware](https://github.com/microsoft/vscode/issues/77790#issuecomment-514106023) of the Clipboard API, and waits patiently. You don't need to ping or bump.
> 
> The specification is managed by W3C, not Mozilla.
> 
> The APIs still have [a long way to go](https://w3c.github.io/clipboard-apis/) before becoming production-ready, and receive very limited [support](https://caniuse.com/mdn-api_clipboard_read) now.

Okay, thank you! I am sorry about my ping or bump, and I will wait for it.

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by jrieken** [comment]
There is still no good support across browsers for this: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard#browser_compatibility

**Comment by tkluck** [comment]
> There is still no good support across browsers for this: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard#browser_compatibility

Some time has passed and this seems pretty well supported by now :partying_face: I'd be interested in this API should it become available, so I'm just posting a gentle bump here.
